### DISCUSSION TITLE
Taroify 轻量、可靠的小程序端 Taro 组件库

### DISCUSSION THREAD

**Discussion by ickg5** [NervJS/taro#10590] [生态]
# Taroify 轻量、可靠的小程序端 Taro 组件库

自 Taro3.0 发布以来，Taro 对 React 和 Vue3 做了比较全面的支持，使得开发者可以使用比较熟悉的 Web 技术开发小程序。由于小程序自身设计的局限性使得大量丰富的 Web UI 组件库不能在小程序上使用，因此我们迫切需要一个丰富的 UI 组件库适配在 Taro 上用以开发小程序。

经过一段时间的调研，Vant 无疑是最流行的移动端组件库之一。最终我们决定将 Vant 移植到 Taro 上，并且采用 React 开发，希望能为 Taro React 带来更有活力的生态。由于 Vant 是使用 Vue3 开发的，它的很多设计理念更接近 Vue3，因此我们对 Taroify 的 API 进行了系统性的设计。Taroify 采用**组合式组件**和 **CSS 优先**的模式设计组件系统，意味着在使用 Taroify 的组件时更加灵活。Taroify 在视觉风格上始终保持与 Vant 一致，这意味着你不用担心需要熟悉另一种设计风格。废话不多说，请开启你的 Taroify 使用之旅吧。

## 介绍

Taroify 是移动端组件库 [Vant](https://github.com/youzan/vant) 的 Taro 版本，两者基于相同的视觉规范，提供近似一致的 API 接口，助力开发者快速搭建小程序应用。

## 特性

- 🚀 性能极佳，组件平均体积小于 1KB（min+gzip）
- 🚀 60+ 个高质量组件，覆盖移动端主流场景
- 💪 使用 TypeScript 编写，提供完整的类型定义
- 💪 单元测试覆盖率超过 90%，提供稳定性保障
- 📖 提供完善的中文文档和组件示例
- 🍭 支持 React
- 🍭 支持主题定制，内置 700+ 个主题变量
- 🍭 支持按需引入和 Tree Shaking
- 🌍 支持国际化和语言包定制
### 通过 npm 安装使用（推荐）

* 通过 npm 安装

```bash
npm i @taroify/core -S --production
```

* 通过 yarn 安装

```bash
yarn add @taroify/core --production
```

## 使用组件

```tsx
import { Button } from "@taroify/core"
import "@taroify/core/button/style"

function Index() {
  return <Button color="primary">按钮</Button>
}
```

更多使用方式，请参考[快速上手](https://taroify.gitee.io/taroify.com/quickstart/)

## 微信交流群

<img src="https://gitee.com/mallfoundry/taroify/raw/main/wechat-qrcode.png" width="200" />
<br />
备注 "taroify" 加好友后邀请进群

## 链接

* [项目地址](https://github.com/mallfoundry/taroify)
* [文档地址](https://taroify.gitee.io/taroify.com/introduce/)
* [更新日志](https://taroify.gitee.io/taroify.com/changelog/)
* [意见反馈](https://github.com/mallfoundry/taroify/issues)

## 未来计划

历时 8 个月，我们最终把 Taroify 基本移植完成。将 Vant Vue3 移植到 Taro React 上耗费了我们大量的精力，最初我以为可能只需要 100 天就能完成，没想到最后用了 8 个月。未来我们希望 Taroify 能在开发、测试以及后续迭代中取得优秀的成绩，更好的服务开发者使用 Taro React 开发小程序。

## 社区力量

* [@Pingren](https://github.com/Pingren)
* [@bingtsingw](https://github.com/bingtsingw)
* [@lblblong](https://github.com/lblblong)
* [@akazwz](https://github.com/akazwz)

特别感谢以上开发者在 Taroify 整个迭代过程中所做的贡献，社区的力量一直在支撑 Taroify 学习和进步。

## 作者介绍

大家好，我是唐治。目前是一名自由开发者，主要从事前后端方面的开发。开发 Taroify 的最初目的是为了提高我们使用 Taro React 的开发效率，同时丰富 Taro React 的生态。现在 Taroify 作为开源项目的一分子，将为 Taro React 开发者提供便捷、轻量、可靠的小程序端组件库。

作为自由开发者，我们是一群充满乐趣、智慧、魅力和极具创意的人。不断追求完美的作品，我们热衷且擅长于小程序开发、Web 开发以及 APP 开发。

你有一个很棒的产品吗？联系我们！

<img src="https://gitee.com/mallfoundry/taroify/raw/main/wechat-qrcode.png" width="200" />
<br />
备注 "项目合作" 开启一个日进斗金的产品！

## 开源协议

本项目基于 [MIT](https://zh.wikipedia.org/wiki/MIT%E8%A8%B1%E5%8F%AF%E8%AD%89) 协议，请自由地享受和参与开源。

**Comment by RoyRao2333** [comment]
请问跟`@antmjs/vantui`相比有什么差异和优势呢？
